src/Language/PureScript/CoreFn/ToJSON.hs

```diff
@@ -0,0 +1,189 @@
+-- |
+-- Dump the core functional representation in JSON format for consumption
+-- by third-party code generators
+--
+module Language.PureScript.CoreFn.ToJSON
+  ( moduleToJSON
+  , annToJSON
+  ) where
+
+import Prelude.Compat
+
+import Data.Aeson
+import Data.Text (pack)
+
+import Language.PureScript.AST.Literals
+import Language.PureScript.Comments
+import Language.PureScript.CoreFn
+import Language.PureScript.Names
+import Language.PureScript.Types
+
+literalToJSON :: (a -> Value) -> Literal a -> Value
+literalToJSON _ (NumericLiteral (Left n)) = toJSON ("NumericLiteral", "Int", n)
```

Comment:
These could just be called `IntLiteral` and `NumberLiteral`. The fact that they're represented using the same constructor in the compiler is just a historical detail, really.